app/src/openxr/cpp/DeviceDelegateOpenXR.cpp

```diff
@@ -1071,13 +1161,23 @@ DeviceDelegateOpenXR::EndFrame(const FrameEndMode aEndMode) {
       return;
   }
 
-  // Add skybox layer
-  if (m.cubeLayer && m.cubeLayer->IsLoaded() && m.cubeLayer->IsDrawRequested()) {
-    m.cubeLayer->Update(m.localSpace, predictedPose, XR_NULL_HANDLE);
-    for (uint32_t i = 0; i < m.cubeLayer->HeaderCount(); ++i) {
-      layers.push_back(m.cubeLayer->Header(i));
+  // Add skybox or passthrough layer
+  if (m.passthroughLayer == nullptr || !m.passthroughLayer->IsDrawRequested() || !m.CanEnablePassthrough()) {
```

Comment:
I think this will be easier to understand if switched to a positive condition, i.e.

```
if (m.passthroughLayer != nullptr && m.passthroughLayer->IsDrawRequested() && m.CanEnablePassthrough()) {
...passthrough stuff
} else if (m.cubeLayer ....) {
... skybox stuff
}
```

Reply:
Right, will fix that.